### ISSUE TITLE
No option to disable GFD from helm chart

### ISSUE THREAD

**Issue by faiq** [NVIDIA/gpu-operator#405]
_The template below is mostly useful for bug reports and support questions. Feel free to remove anything which doesn't apply to you and add more information where it makes sense._

### 1. Quick Debug Checklist
- [ ] Are you running on an Ubuntu 18.04 node?
- [ ] Are you running Kubernetes v1.13+?
- [ ] Are you running Docker (>= 18.06) or CRIO (>= 1.13+)?
- [ ] Do you have `i2c_core` and `ipmi_msghandler` loaded on the nodes?
- [ ] Did you apply the CRD (`kubectl describe clusterpolicies --all-namespaces`)

### 1. Issue or feature description
There's no option to disable gfd from the helm chart and make it apply on the clusterPolicy object that is created.

For example the following command 
```
$ helm install --wait --generate-name  --set nfd.enabled=false --set driver.enabled=false --set toolkit.version=v1.10.0-centos7 --set gfd.enabled=false  nvidia/gpu-operator
```
Does not disable the option for the clusterPolicy object 

```
$ kubectl get clusterpolicy -o yaml | grep -A10 gfd
    gfd:
      env:
      - name: GFD_SLEEP_INTERVAL
        value: 60s
      - name: GFD_FAIL_ON_INIT_ERROR
        value: "true"
      image: gpu-feature-discovery
      imagePullPolicy: IfNotPresent
      repository: nvcr.io/nvidia
      version: v0.6.1-ubi8
 ``` 
 
 and the pods are unnecessarily created
 
 ```
 $ kubectl get po
NAME                                       READY   STATUS                  RESTARTS      AGE
gpu-feature-discovery-54vbc                1/1     Running                 0             2m3s
gpu-operator-7c4f4f7b54-z7xlr              1/1     Running                 0             22m
``` 
 
When I pulled the helm release using 

```
helm pull nvidia/gpu-operator
```
and untared it 

```
 tar -xzf gpu-operator-v1.11.1.tgz 
 ```

there is no option for `enabled` 

```
 cat gpu-operator/templates/clusterpolicy.yaml | grep -A10 gfd 
  gfd:
    {{- if .Values.gfd.repository }}
    repository: {{ .Values.gfd.repository }}
    {{- end }}
    {{- if .Values.gfd.image }}
    image: {{ .Values.gfd.image }}
    {{- end }}
    {{- if .Values.gfd.version }}
    version: {{ .Values.gfd.version | quote }}
    {{- end }}
    {{- if .Values.gfd.imagePullPolicy }}
    imagePullPolicy: {{ .Values.gfd.imagePullPolicy }}
    {{- end }}
    {{- if .Values.gfd.imagePullSecrets }}
    imagePullSecrets: {{ toYaml .Values.gfd.imagePullSecrets | nindent 6 }}
    {{- end }}
    {{- if .Values.gfd.resources }}
    resources: {{ toYaml .Values.gfd.resources | nindent 6 }}
    {{- end }}
    {{- if .Values.gfd.env }}
    env: {{ toYaml .Values.gfd.env | nindent 6 }}
    {{- end }}
    {{- if .Values.gfd.args }}
    args: {{ toYaml .Values.gfd.args | nindent 6 }}
    {{- end }}
  migManager:
    enabled: {{ .Values.migManager.enabled }}
    {{- if .Values.migManager.repository }}
    repository: {{ .Values.migManager.repository }}
    {{- end }}
    {{- if .Values.migManager.image }}
    image: {{ .Values.migManager.image }}
    {{- end }}
    {{- if .Values.migManager.version }}
```
### 2. Steps to reproduce the issue

to view the helm chart do `help pull nvidia/gpu-operator` and view the files in `gpu-operator/templates/clusterpolicy.yaml` 

Thanks

### 3. Information to [attach](https://help.github.com/articles/file-attachments-on-issues-and-pull-requests/) (optional if deemed irrelevant)

 - [ ] kubernetes pods status: `kubectl get pods --all-namespaces`
 - [ ] kubernetes daemonset status: `kubectl get ds --all-namespaces`
 - [ ] If a pod/ds is in an error state or pending state `kubectl describe pod -n NAMESPACE POD_NAME`
 - [ ] If a pod/ds is in an error state or pending state `kubectl logs -n NAMESPACE POD_NAME`

 - [ ] Output of running a container on the GPU machine: `docker run -it alpine echo foo`
 - [ ] Docker configuration file: `cat /etc/docker/daemon.json`
 - [ ] Docker runtime configuration: `docker info | grep runtime`

 - [ ] NVIDIA shared directory: `ls -la /run/nvidia`
 - [ ] NVIDIA packages directory: `ls -la /usr/local/nvidia/toolkit`
 - [ ] NVIDIA driver directory: `ls -la /run/nvidia/driver`
 - [ ] kubelet logs `journalctl -u kubelet > kubelet.logs`

**Comment by cdesiniotis** [comment]
The option to disable GFD will be available in the next release. Here is the change: https://github.com/NVIDIA/gpu-operator/commit/976cd11e128a89d429e289d57089d142393e14ba

**Comment by faiq** [comment]
This change isn't available in the 1.11.1 release yet?

**Comment by cdesiniotis** [comment]
It is not included in the 1.11.1 release.

**Comment by cdesiniotis** [comment]
GPU Operator v22.9.0 has been released: https://github.com/NVIDIA/gpu-operator/releases/tag/v22.9.0

You can now disable GFD by setting `gfd.enabled=false` in ClusterPolicy
